### DISCUSSION TITLE
请求根据响应数据缓存功能

### DISCUSSION THREAD

**Discussion by miniocean404** [alovajs/alova#615] [Q&A]
请问是不是 alova 不支持类似 swr 请求的下图中的内容，文档看遍了也没看到合适的方式
![image](https://github.com/user-attachments/assets/ffd1e0f2-74f3-422d-81f5-8a968e37353b)

**Comment by MeetinaXD** [comment]
看文档：[客户端恢复模式](https://alova.js.org/zh-CN/tutorial/cache/mode#客户端-1)

**Comment by miniocean404** [comment]
我看过了，但是哪怕是 https://alova.js.org/zh-CN/tutorial/advanced/in-depth/custom-method-key 这个也不能使用，因为我去想拿到响应来设置 key 的时候 ts 类型就报错，并且运行起来也是有问题的

**Comment by MeetinaXD** [comment]
不知道你具体是怎么用的，但 cache 是 alova 的基本功能。

你可以看看以下示例：

在 [CodeSandbox](https://codesandbox.io/p/sandbox/github/alovajs/alova/tree/main/examples/vue) 中查看RestoreCache 示例。
源码 [RestoreCache.vue](https://github.com/alovajs/alova/blob/main/examples/vue/src/views/RestoreCache.vue)

如果你确定这是一个bug，请将discussion转为一个issue，并提供可复现的项目地址。

**Comment by miniocean404** [comment]
你可能没有理解我的意思，它不像是一个 bug 更像是一个功能
现象：我指的是比如我请求接口了，它将响应缓存了下来，下一次请求调用的时候直接使用了缓存的数据，并没有发送请求。
希望：我请求接口了，下一次继续发送请求，先展示之前的缓存的数据，但是 alova 会对比两次响应的数据，如果不一样则立马替换更新。如果一样则不更新

**Reply by wintsa123** [reply]
奇怪的需求，你的需求好像就是原生axios加pinia啊。。。
你要用alova实现也简单。。。设置不缓存，自己套个pinia不就是了。

**Reply by miniocean404** [reply]
> 奇怪的需求，你的需求好像就是原生axios加pinia啊。。。 你要用alova实现也简单。。。设置不缓存，自己套个pinia不就是了。

我使用的是 taro react 版本，swr 就实现了这个功能，目的是为了优化界面第一次看到白屏等待的问题

**Reply by JOU-amjs** [reply]
@MiniOcean404 可以的，两种方式：

1. 使用`initialData`+`onSuccess`：
https://github.com/alovajs/alova/blob/main/examples/react/src/views/OldDataPlaceholder/Approach1.jsx

2. 封装成`midleware`，扩展性更高：
https://github.com/alovajs/alova/blob/main/examples/react/src/views/OldDataPlaceholder/Approach2.jsx

**Comment by wintsa123** [comment]
其实如果你熟悉alovajs的话，你也可以做到这一点。alovajs发送请求是预留了一个可以直接显示的值的。你只需要用缓存管理把旧的缓存拿出来即可

小海洋 ***@***.***> 于2024年12月27日周五 15:59写道：

**Comment by miniocean404** [comment]
非常感谢大家的帮助
